docs/architecture/event-flows-proposal.md

```diff
@@ -0,0 +1,77 @@
+# Eventing flow proposal
+
+## Overview
+
+This document shows a proposal of two ways of eventing flow we want to support with use of the [Management Plane](./terminology.md#management-plane)
+
+## Eventing with Kyma internal Event Bus
+
+![Management Plane Components](assets/internal-event-flow.svg)
+
+1. The event is registered using **Management Plane** API.
+2. The **Director** sends an event url to Application.
+3. The **Agent** fetches the event data and passes it to Kyma Event Bus.
+4. The Application publishes an event.
+
+## Eventing with external Event Bus
+
+![Management Plane Components](assets/external-event-flow.svg)
+
+1. The event is registered using **Management Plane** API with external event bus data.
+2. The **Agent** fetches the event and passes information to eventing system (E.g Google's Cloud PubSub).
+3. Eventing system subscribes on external event bus.
+4. The application publishes an event.
+
+## GraphQL Schema  
+
+### Types
+```graphql
+
+type EventAPIDefinition {
```

Comment:
take into account my comment from Event Gateway and External Bus. Make possible, that application can setups EventGateway or provide credentials for External Bus.